### PR TITLE
Fix calling UnknownCommandEvent

### LINKED ISSUE #8225: UnknownCommandEvent only gets fired when the console uses an unknown command
### Expected behavior

For the event to be fired not only when the console uses an unknown command

### Observed/Actual behavior

The event is only fired when the console uses an unknown command

### Console
(ignore the debug print to stdout)
Console
### Player
Player

### Plugin and Datapack List

Only one plugin which i made
The repo: https://github.com/Skulk-Network/Plugin
(related: https://github.com/Skulk-Network/Plugin/blob/master/src/main/java/network/skulk/plugin/extensions/message_override/listeners/UnknownCommandListener.java)

### Paper version

```
This server is running Paper version git-Paper-102 (MC: 1.19.1) (Implementing API version 1.19.1-R0.1-SNAPSHOT) (Git: e0486ae)
You are running the latest version
Previous version: git-Paper-99 (MC: 1.19.1)
```

### Other

_No response_

### LINKED ISSUE #8329: papers config no-permission message comes back a different message than the one you set
### Expected behavior

doing a command you dont have permission for  should be the message you set in papers config

also running an unknown command should be what is set in spigots config

### Observed/Actual behavior

the message sent is "Uknown or incomplete command, see below for error"

the message sent for an unknown command is also "Uknown or incomplete command, see below for error"

### Steps/models to reproduce

run an unknown command or a command you dont have permission for as a player in a 1.19.2 paper server. 

it sends the configured message when you run an unknown command in the console.

### Plugin and Datapack List

none

### Paper version

paper-1.19.2-132

### Other

this problem only happens on a paper 1.19.2 server, 1.19 does not do this

### PR DESCRIPTION
**Pull request by Machine-Maker**
Fixes https://github.com/PaperMC/Paper/issues/8225
Fixes https://github.com/PaperMC/Paper/issues/8329

### CODE REVIEW

**Review by kennytv — DISMISSED** [review]
`/tp` needs to have the brig message passed into the event, should also add a small note that this now includes invalid command syntax as unknown commands for called events

File: patches/server/0139-Add-UnknownCommandEvent.patch [hunk 1]
```diff
@@ -3,9 +3,108 @@ From: Sweepyoface <github@sweepy.pw>
...
+                 }
++                // Paper end
+ 
+                 b1 = 0;
+                 return b1;
+@@ -350,6 +378,7 @@ public class Commands {
+                 b0 = 0;
+             }
+         } finally {
++            this.dispatcher.setConsumer(existing); // Paper
```
Comment (electronicboy) [inline patches/server/0139-Add-UnknownCommandEvent.patch]: I would probably validate that the consumer hasn't changed in the meantime

**Review by electronicboy — APPROVED** [review]

File: patches/server/0139-Add-UnknownCommandEvent.patch [hunk 2]
```diff
@@ -3,9 +3,108 @@ From: Sweepyoface <github@sweepy.pw>
...
++++ b/src/main/java/net/minecraft/commands/CommandSourceStack.java
+@@ -329,8 +329,13 @@ public class CommandSourceStack implements SharedSuggestionProvider {
+     }
+ 
+     public void sendFailure(Component message) {
++        // Paper start
++        this.sendFailure(message, true);
++    }
++    public void sendFailure(Component message, boolean withStyle) {
++        // Pape rend
```
Comment (Lulu13022002) [inline patches/server/0139-Add-UnknownCommandEvent.patch]: Small typo

File: patches/server/0139-Add-UnknownCommandEvent.patch [hunk 3]
```diff
@@ -3,9 +3,108 @@ From: Sweepyoface <github@sweepy.pw>
...
++                        .append(net.kyori.adventure.text.Component.newline())
++                        .append(io.papermc.paper.adventure.PaperAdventure.asAdventure(ichatmutablecomponent));
++                }
++                }
++                org.bukkit.event.command.UnknownCommandEvent event = new org.bukkit.event.command.UnknownCommandEvent(commandlistenerwrapper.getBukkitSender(), s, builder.build());
++                org.bukkit.Bukkit.getServer().getPluginManager().callEvent(event);
++                if (event.message() != null) {
++                    commandlistenerwrapper.sendFailure(io.papermc.paper.adventure.PaperAdventure.asVanilla(event.message()), false);
+                 }
++                // Paper end
```
Comment (Lulu13022002) [inline patches/server/0139-Add-UnknownCommandEvent.patch]: This comment should probably go one line above

File: patches/server/0139-Add-UnknownCommandEvent.patch (hunk 1 above)
Comment (Machine-Maker) [reply patches/server/0139-Add-UnknownCommandEvent.patch]: I just removed the whole consumer thing. Found a better way to know if a vanilla command was run

**Comment by pauhull** [comment]
Is this going to be fixed?

**Review by electronicboy — APPROVED** [review]

**Pull request merged**